File: tests/benchmark/mapfold/ocaml/mapfold.ml
(* map and fold a list of N integers K times *)

let rec run n acc lst =
  if n == 0 then
    acc
  else
    run (n - 1) (List.fold_left (+) (-acc) lst) (List.map ((+) 1) lst)

let rec gen k n =
  if k == n then
    [k]
  else
    k :: gen (k + 1) n

;;

let k = 10000
and n = 10000
in

print_int (run k 0 (gen 1 n));
print_newline ()

File: tests/benchmark/mapfold/haskell/mapfold.hs
-- map and fold a list of N integers K times

import Data.List

run :: Int -> Int -> [Int] -> Int
run 0 acc lst = acc
run n acc lst = run (n - 1) (foldl' (+) (-acc) lst) (map (+ 1) lst)

main :: IO ()
main = putStrLn (show (run k 0 [1..n]))
  where
    k = 10000
    n = 10000
